kore/src/Kore/Unification/Procedure.hs

```diff
@@ -71,8 +77,16 @@ unificationProcedure sideCondition p1 p2
         orCeil <- liftSimplifier $ makeEvaluateTermCeil sideCondition term
         marker "unify" "CombineCeil"
         ceil' <- Monad.Unify.scatter orCeil
-        lowerLogicT . simplifyCondition sideCondition $
-            Conditional.andCondition ceil' condition
+        condition' <-
+            lowerLogicT . simplifyCondition sideCondition $
+                Conditional.andCondition ceil' condition
+        when (condition /= condition') $
+            logError $
+                "Simplification did something.\n New unifier produced:\n"
+                    <> pack (show condition)
+                    <> "Then simplificaton produced:\n"
+                    <> pack (show condition')
```

Comment:
Just stumbled over this... might be better to use the `Pretty` printer than `show`. There are a few utilities in https://github.com/runtimeverification/haskell-backend/blob/master/kore/src/Pretty.hs